misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/completion/FunctionCompletionItemBuilder.java

```diff
@@ -0,0 +1,151 @@
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.debugadapter.completion;
+
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.debugadapter.completion.context.CompletionContext;
+import org.ballerinalang.debugadapter.completion.util.CommonUtil;
+import org.eclipse.lsp4j.debug.CompletionItem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is being used to build function type completion item.
+ *
+ * @since 2201.1.0
+ */
+public final class FunctionCompletionItemBuilder {
+
+    private static final String PARENTHESIS = "()";
+    private static final String OPEN_PARENTHESIS = "(";
+    private static final String CLOSE_PARENTHESIS = ")";
+
+    private FunctionCompletionItemBuilder() {
+    }
+
+    /**
+     * Creates and returns a completion item.
+     *
+     * @param functionSymbol BSymbol
+     * @param context        Debug completion Context
+     * @return {@link CompletionItem}
+     */
+    public static CompletionItem build(FunctionSymbol functionSymbol, CompletionContext context) {
+        CompletionItem item = new CompletionItem();
+        if (functionSymbol != null && functionSymbol.getName().isPresent()) {
+            String funcName = functionSymbol.getName().get();
+            Pair<String, String> functionSignature = getFunctionInvocationSignature(functionSymbol, funcName, context);
+            item.setText(functionSignature.getLeft());
+            item.setLabel(functionSignature.getRight());
+        }
+        return item;
+    }
+
+    /**
+     * Get the function invocation signature.
+     *
+     * @param functionSymbol Ballerina function instance
+     * @param functionName   Function name
+     * @param context        Debug completion Context
+     * @return {@link Pair} of insert text(left-side) and signature label(right-side)
+     */
+    private static Pair<String, String> getFunctionInvocationSignature(FunctionSymbol functionSymbol,
+                                                                       String functionName,
+                                                                       CompletionContext context) {
+        if (functionSymbol == null) {
```

Comment:
Should we consider the nullity here? If the function symbol is null isn't that an invalid scenario?